app/covidbed/repository/resource.py

```diff
@@ -0,0 +1,28 @@
+from datetime import datetime
+from covidbed.model import Organization, Platform, Contact, Availability
+
+
+def create_availability(platform, organization, date, functional_unit=None, bed=None, supply=None, human=None, contact=None):
+    assert isinstance(platform, Platform)
+    assert isinstance(organization, Organization)
+    assert isinstance(date, datetime)
+    assert bed is None or isinstance(bed, dict)
+    assert supply is None or isinstance(supply, dict)
+    assert human is None or isinstance(human, dict)
+    assert contact is None or isinstance(contact, dict) or isinstance(contact, Contact)
+
+    if isinstance(contact, dict):
+        contact_obj = Contact(**contact)
```

Comment:
Shouldn't we first check if the contact is already in the db?

Reply:
done y the upper layer (api ) where check is done. Here we check only if the param is a dict (creation) or a model (existed one)

Reply:
Ok!